packages/swing-store/src/transcriptStore.js

```diff
@@ -516,55 +520,109 @@ export function makeTranscriptStore(
     noteExport(spanMetadataKey(rec), JSON.stringify(rec));
   };
 
+  function addTranscriptSpanRecord(metadata) {
+    const { vatID, startPos, endPos, hash, isCurrent, incarnation } = metadata;
+    vatID || Fail`transcript metadata missing vatID: ${metadata}`;
+    startPos !== undefined ||
+      Fail`transcript metadata missing startPos: ${metadata}`;
+    endPos !== undefined ||
+      Fail`transcript metadata missing endPos: ${metadata}`;
+    hash || Fail`transcript metadata missing hash: ${metadata}`;
+    isCurrent !== undefined ||
+      Fail`transcript metadata missing isCurrent: ${metadata}`;
+    incarnation !== undefined ||
+      Fail`transcript metadata missing incarnation: ${metadata}`;
+
+    sqlWriteSpan.run(
+      vatID,
+      startPos,
+      endPos,
+      hash,
+      isCurrent ? 1 : null,
+      incarnation,
+    );
+  }
+
+  const sqlGetSpanMetadataFor = db.prepare(`
+    SELECT hash, isCurrent, incarnation
+      FROM transcriptSpans
+      WHERE vatID = ? AND startPos = ? AND endPos = ?
```

Comment:
I feel like it'd be more a more versatile query if the `endPos` was selected, and not part of the condition.

Reply:
Hm, maybe. What I need it for is to see if we have a validation hash for a particular span artifact (in `populateTranscriptSpan`), which has a specific `endPos` in mind. It's also used by `repairTranscriptSpanRecord` (to see if the record already exists).

Changing it to return `endPos` would need an `assert.equal` in both places. In `populateTranscriptSpan`, a mismatch would want to emit the same "no metadata for that thing" error that the zero-row return would provoke. For `repairTranscriptSpanRecord`, I guess it would fare better in the error case where the DB already has a span record for deliveries `4..8`, but the repairing metadata has one for `4..9` (for the same vatID). My version would think that the `4..9` record is new, and try to insert it, which would fail the PRIMARY KEY uniqueness constraint. An explicit check would be able to throw a "metadata mismatch" error.

If "more versatile" is the only improvement, I'm not sure it's worth a change. All the use cases have a specific `endPos` in mind, so it felt more natural to include it in the query. But I'd accept an argument that it would catch more errors or report errors better.

Reply:
I think `populateTranscriptSpan` should skip any transcript artifact that contains less than the number of entries dictated by the local span. It could use artifacts that have more entries than the span by ignoring the extra lines, but only for the current span.
That would support using artifacts that are both newer and older to populate transcript spans

Reply:
I'll switch this query to return `endPos` instead of taking it.

`populateTranscriptSpan` will throw if `endPos` does not match. I don't think I want it trying to utilize mismatched spans, even if the supplied one is larger than expected (and thus supposedly has all the items it needs, plus more). There's no good reason that they would not match: historical spans have fixed+known bounds, and the current span must not be pruned.

Reply:
addressed in 4020cac69a004ea274b46158e83675b8e59fd0f7